proxy/proxy.go

```diff
@@ -34,6 +40,40 @@ func formatRequest(r *http.Request) string {
 	return strings.Join(request, "\n")
 }
 
+func pollConfigChange(config *Config) {
+	//we check for file changes every half seccond
+	ticker := time.NewTicker(500 * time.Millisecond)
+	quit = make(chan struct{})
+	go func() {
+		for {
+			select {
+			case <-ticker.C:
+				info, err := os.Stat(configFile)
+				if err != nil {
+					log.Printf("Error reading config file: %s\r\n", err.Error())
+					continue
+				}
+
+				if info.ModTime().Before(modTime) || info.Size() != size {
+					services, err := LoadServices(configFile)
+					if err != nil {
+						log.Printf("Error reading the modified config file: %s\r\n", err.Error())
+						continue
+					}
+					//we must lock to be sure that we are thread safe when loading the new config
+					mu.Lock()
```

Comment:
I know those locks are necessary when sharing memory through channels, but I was thinking if isn't there a way to avoid them? 
Maybe trying the approach of share memory by communication: https://blog.golang.org/share-memory-by-communicating